### PR TITLE
[ENH] Silhouette: Add report, improve GUI

### PR DESCRIPTION
**Pull request by janezd**

### CODE REVIEW

File: Orange/widgets/unsupervised/owsilhouetteplot.py [hunk 1]
```diff
@@ -308,6 +317,11 @@ def commit(self):
         self.send("Selected Data", selected)
         self.send("Other Data", other)
 
+    def send_report(self):
+        self.report_plot()
+        self.report_caption("Silhouette plot with ({} distance)".
```
Comment (lanzagar) [inline Orange/widgets/unsupervised/owsilhouetteplot.py]: Why is the second part of this sentence in parentheses?
I think either the word "with" or the parentheses need to be removed.

**Comment by lanzagar** [comment]
The variable which was used for grouping is not reported!

**Comment by lanzagar** [comment]
I am guessing the annotation var should also be reported, but could not even try it out - do annotations work at all?
The combo does not include discrete variables, and string variables are not shown in the plot.

This could be a bug fix for another PR or added here.

**Comment by janezd** [comment]
I omitted the names of the clustering and annotation variables thinking that these would be obvious from the values appearing in the graph. But they're probably not, so I added them to the report.

Annotation needs string meta variables. You can try them with zoo, except that you have to continuize it first.

A matter for another PS is that the plot is scaled to some fixed height, I guess. If you increase the bar widths, the labels get smaller.

**Comment by lanzagar** [comment]
currently: clustered by 'type', annotated by 'name'
better: grouped by 'type', annotated with 'name'

Other than that I think the report is now OK.

As for the annotations - I almost wrote that I tried on zoo and they did not work. However I discovered the problem. If bar width is less than 5 (default 3) they do not appear. Without any indication or explanation. Very confusing and misleading. But probably to be handled in another PR?

**Comment by janezd** [comment]
by -> with: yes, sure.

I added the indication in a separate commit (but this same PR). I didn't use `self.warning` since the move the interface, pushing the slider from below the mouse. The label also looks better, and there is enough space for it, too.

**Comment by lanzagar** [comment]
I like the label.
@janezd can you take care of the CLA (maybe you don't have all emails you're using added to the github account?) so this can be merged.

**Comment by janezd** [comment]
I was using the same mail. I pushed the last commit again; now it's OK.

**Pull request merged**